### DISCUSSION TITLE
Nuxt 4: auto import for stores breaks

### DISCUSSION THREAD

**Discussion by onurusluca** [nuxt/nuxt#28982] [Questions] [answered]
I just upgraded to Nuxt 4 as per here: https://nuxt.com/docs/getting-started/upgrade#upgrading-nuxt
Everything except store auto imports works fine.
I used to work with no problem at all.

/stores/general.store.ts:
```
import { defineStore } from "pinia";
import type { VendorProfile } from "@@/types/supabase.types.js";

export const useGeneralStore = defineStore("generalStore", {
  state: () => ({
    vendorProfile: null as VendorProfile | null,
  }),
  getters: {},
  actions: {},
});
```

What am I doing wrong. I couln't find any explanation anywhere about this.

app/components/SomeComponent.vue;
`const generalStore = useGeneralStore();` _(Cannot find name 'useGeneralStore'.)_

Deleted .nuxt, node_modules, lock file, cleared all caches, updated to latest. Nothing worked.

**Comment by onurusluca** [comment, marked as answer]
Of course just after 5 minutes of posting the question, I find the answer. :D

Fix:
```
  pinia: {
    storesDirs: ["./stores/**"],
  },
```

This tells Nuxt where to look for the stores to auto import. The thing is that I didn't move the folder anywhere.
Did Nuxt forget? Is this maybe a bug? I am not really at liberty to say. Can someone confirm?

Also, there is no mention to change the imports on upgrading guide.
I had to change some of my imports that are outside of the "app" folder because now `@` or `~` not points to "app" not "/". 
Is this also a problem? Should I open a PR or something? I don't know. Anyone more knowledgeable than me?

**Reply by maxdzin** [reply]
Hi @onurusluca 

>This tells Nuxt where to look for the stores to auto import. The thing is that I didn't move the folder anywhere.
Did Nuxt forget? Is this maybe a bug? I am not really at liberty to say. Can someone confirm?

I think that `stores` are not moved because they are assumed or intended to be used on both client and server side.

>I had to change some of my imports that are outside of the "app" folder because now @ or ~ not points to "app" not "/".
Is this also a problem? Should I open a PR or something? I don't know. Anyone more knowledgeable than me?

With a new directory structure, `~` and `@` are pointing to app, while `~~` and `@@` are pointing to the project root.
It seems that will be mentioned soon in a more comprehensive guide with Nuxt 4 release.

**Reply by maxdzin** [reply]
As for those aliases, those are mentioned in docs here, in the config section:
https://nuxt.com/docs/api/nuxt-config#alias
```ts
{
  "~": "/<srcDir>",
  "@": "/<srcDir>",
  "~~": "/<rootDir>",
  "@@": "/<rootDir>",
  "assets": "/<srcDir>/assets",
  "public": "/<srcDir>/public"
}
```
and, as for the new v4 directory structure, `srcDir` is now `app`.
